### ISSUE TITLE
Sometimes, computer karts explode after a restart

### ISSUE THREAD

**Issue by Wuzzy2** [supertuxkart/stk-code#3820]
## Description
Singleplayer mode: If you restart a track, sometimes one of the computer-controlled karts starts with the damage stars above the head and when the race starts (green lamp), the kart explodes.

## Steps to reproduce
Play singleplayer and restart a race. Sometimes, a kart has the stars.

I don't know exactly when this happens but I suspect this happens when you restart a race while a kart is exploding, but I have not verified this.

## Configuration
STK release version: 0.10-rc1
STK Source (PPA, distribution package, official binary, etc.): Git

**Comment by Alayan-stk-2** [comment]
A method to reproduce this reliably would be useful.

I will do some tests, it sounds like the reset function isn't working as expected in your situation.

**Comment by Alayan-stk-2** [comment]
I did some attempts to reproduce this. Sometimes (but not always), when restarting the race while another kart has an explosion effect, an explosion will be visible after the karts have been placed back to the start position.

But I found something else which has a 100% reproduction rate and may actually be the only issue : when a kart has a bomb on when the restart happen, the explosion is shown after restart.

This is a purely visual issue : the karts are properly placed and don't move.
